### DISCUSSION TITLE
Windows usb device not recognized solution

### DISCUSSION THREAD

**Discussion by cjs30** [micropython/micropython#10458] [RP2040 / Pico]
I have had a persistent but intermittent USB device not recognized when plugging in a Pico to a Windows laptop which was causing an issue with Thonny etc. I tried all of the solutions I could find online.

The pico has a main.py on it. The solution - which so far seems to be robust - was to simply add a time.sleep(1) at the start of the main.py. I haven't worked out what elements in the main.py stop Windows recognizing the device but it does seem that the 'processor' is involved in negotiating the device driver used.

I wonder if it might be possible to add a delay - within the micropython firmware - that enforces the delay before main.py starts? I suspect that 1s is overkill. It might prevent such problems. Perhaps this is already well known and my laptop is just a bit slow at negotiating the device driver selection...

**Comment by robert-hh** [comment]
I have noticed that as well. See PR #10423. Happens with Linux as well, and also with other boards & ports.

**Reply by dpgeorge** [reply]
This should be fixed by 5890a17ae049aad39f97744cadf5844c06a5a4c0
